src/spid/spid.ts

```diff
@@ -44,6 +48,28 @@ import {
 } from "../utils/saml";
 import { getMetadataTamperer } from "../utils/saml";
 
+export const SpidUser = t.intersection([
```

Comment:
Just a reminder: this type in my opinion should be autogenerated from an `openapi` spec, but such refactoring could be out of scope for this PR

Reply:
Maybe we can't do that for `getAssertionXml: t.Function`

Reply:
@BurnedMarshal I agree, but this `type` is not used to define the SAML Strategy, but only to decode a user payload. As far as I can understand, this line could be commented out and the SPID login should still work. Maybe I'm wrong, as I don't know very well the inner workings of `io-spid-common`

Reply:
@GiovanniMancini if you see inside [io-backend](https://github.com/pagopa/io-backend/blob/master/src/controllers/authenticationController.ts#L81) the value that comes from the strategy is not the value stored inside redis. Before happening a validation with `validateSpidUser` and after with `toAppUser` we have the value that should be stored.

Reply:
@BurnedMarshal As far as I can understand, in `io-backend` the method `getAssertionXml` is used inside `validateSpidUser` to check if `spidLevel` is `L2` (and set it to `L2` if it isn't set). Do you think we should do the same thing here?

Reply:
Reading from the comment the default `L2` is for a bug of `spid-testenv`. Another operation done by `validateSpidUser` is the prefix removal from`fiscal_code` field.

Reply:
@BurnedMarshal  OK, let's keep the method, but that means we should change the way we parse user payload and adhere in a more restrictive way to what is done in `io-backend`. Maybe we can do that as a part of another PR